src/linker/Linker/Annotations.cs

```diff
@@ -49,9 +49,11 @@ public partial class AnnotationStore
 		protected readonly Dictionary<FieldDefinition, object> field_values = new Dictionary<FieldDefinition, object> ();
 		protected readonly HashSet<FieldDefinition> field_init = new HashSet<FieldDefinition> ();
 		protected readonly HashSet<TypeDefinition> fieldType_init = new HashSet<TypeDefinition> ();
-		protected readonly HashSet<IMetadataTokenProvider> marked = new HashSet<IMetadataTokenProvider> ();
+
+		protected readonly HashSet<IMetadataTokenProvider> marked_pending = new HashSet<IMetadataTokenProvider> ();
```

Comment:
I think it would be worth adding a short comment describing what this field stores and what it's for... as there's additional complexity to these, the code is not self-descriptive enough I think

Reply:
Added a comment